src/app/printerprofile.service.ts

```diff
@@ -0,0 +1,69 @@
+import { HttpClient, HttpErrorResponse } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import { Subscription } from "rxjs";
+
+import { ConfigService } from "./config/config.service";
+import { NotificationService } from "./notification/notification.service";
+import { PrinterService } from "./printer.service";
+
+import { OctoprintPrinterProfileAPI } from "./octoprint-api/printerProfileAPI";
+
+@Injectable({
+  providedIn: "root",
+})
+export class PrinterProfileService {
+  private httpGETRequest: Subscription;
+  
+  public constructor(
+    private http: HttpClient,
+    private configService: ConfigService,
+    private notificationService: NotificationService,
+    private printerStatusService: PrinterService,
+    private router: Router
+  ) { }
+
+  public getDefaultPrinterProfile(): Promise<OctoprintPrinterProfileAPI> {
+    return new Promise((resolve): void => {
+      if (this.httpGETRequest) {
+        this.httpGETRequest.unsubscribe();
+      }
+      this.httpGETRequest = this.http
+        .get(this.configService.getURL("printerprofiles/_default"), this.configService.getHTTPHeaders())
+        .subscribe(
+          (data: OctoprintPrinterProfileAPI): void => {
+            const printerProfile: OctoprintPrinterProfileAPI = data;
+            resolve(printerProfile);
```

Comment:
```suggestion
          (printerProfile: OctoprintPrinterProfileAPI): void => {
            resolve(printerProfile);
```

The observable already maps to the correct type, so we can save that variable as far as I can tell.